### PR TITLE
Use Url for Tracker.url, update deps

### PR DESCRIPTION
**Pull request by ParadoxSpiral**
<s>The `let url = …` is a bit ugly, but I dunno how to do this better, since we can't use `.map(|u| u.as_str()).unwrap_or("")` because that would reference a tmp str.</s>

### CODE REVIEW

File: rpc/src/resource.rs [hunk 1]
```diff
@@ -845,10 +848,15 @@ impl Queryable for Peer {
...
+        let url = if let Some(ref u) = self.url {
+            u.as_str()
+        } else {
+            ""
+        };
         match f {
             "id" => Some(Field::S(&self.id)),
             "torrent_id" => Some(Field::S(&self.torrent_id)),
-            "url" => Some(Field::S(&self.url)),
+            "url" => Some(Field::S(url)),
```
Comment (Luminarys) [inline rpc/src/resource.rs]: Use `Field::O` here instead of that `let url` bit.

File: src/torrent/info.rs [hunk 2]
```diff
@@ -107,10 +109,12 @@ impl Info {
                 })
             })
             .ok_or("No hash found in magnet")?;
-        let announce = url.query_pairs()
-            .find(|&(ref k, _)| k == "tr")
-            .map(|(_, ref v)| v.to_string())
-            .unwrap_or_else(|| "".to_owned());
+        let announce = url.query_pairs().find(|&(ref k, _)| k == "tr").and_then(
+            |(_,
+              ref v)| {
```
Comment (Luminarys) [inline src/torrent/info.rs:114]: This and the previous line should be combined.

File: src/torrent/info.rs [hunk 3]
```diff
@@ -107,10 +109,12 @@ impl Info {
...
+                Url::parse(&v).map(|u| Some(u)).unwrap_or(None)
```
Comment (Luminarys) [inline src/torrent/info.rs]: Is `&v` necessary with the `ref v`?

File: src/torrent/info.rs (hunk 2 above)
Comment (ParadoxSpiral) [reply src/torrent/info.rs:114]: The version of rustfmt on travis doesn't like that. https://travis-ci.org/Luminarys/synapse/jobs/317495779#L491

**Comment by Luminarys** [comment]
Thanks!

**Pull request merged**